test/unit/ipam_test.rb

```diff
@@ -0,0 +1,90 @@
+                                  :ipam => IPAM::MODES[:db])
+      ipam = IPAM::Db.new(:subnet => subnet)
+      assert_equal '192.168.2.1', ipam.suggest_ip
+
+      subnet.reload
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.1', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:01')
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.2', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:02')
+      assert_equal '192.168.2.3', ipam.suggest_ip
+    end
+
+    test "should find unused IP excluding named values in internal DB if proxy is not set" do
+      host = FactoryGirl.create(:host)
+      subnet = FactoryGirl.create(:subnet_ipv4, :name => 'my_subnet', :network => '192.168.2.0',
+                                  :ipam => IPAM::MODES[:db])
+      ipam = IPAM::Db.new(:subnet => subnet, :excluded_ips => ['192.168.2.1', '192.168.2.2'])
+      assert_equal '192.168.2.3', ipam.suggest_ip
+
+      subnet.reload
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.1', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:01')
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.2', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:02')
+      ipam = IPAM::Db.new(:subnet => subnet, :excluded_ips => ['192.168.2.3'])
+      assert_equal '192.168.2.4', ipam.suggest_ip
+    end
+
+    test "should respect subnet from and to if it's set" do
+      host = FactoryGirl.create(:host)
+      subnet = FactoryGirl.create(:subnet_ipv4, :name => 'my_subnet', :network => '192.168.2.0', :from => '192.168.2.10', :to => '192.168.2.12',
+                                  :ipam => IPAM::MODES[:db])
+      ipam = IPAM::Db.new(:subnet => subnet)
+      assert_equal '192.168.2.10', ipam.suggest_ip
+
+      subnet.reload
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.10', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:01')
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.11', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:02')
+      assert_equal '192.168.2.12', ipam.suggest_ip
+
+      subnet.reload
+      FactoryGirl.create(:nic_managed, :ip => '192.168.2.12', :subnet_id => subnet.id, :host => host, :mac => '00:00:00:00:00:03')
+      assert_nil ipam.suggest_ip
+    end
+  end
+
+  context 'EUI-64 IPAM' do
+    test "should calculate unused IP via eui-64" do
+      subnet = FactoryGirl.build(:subnet_ipv6,
+                                 :network => '2001:db8::',
+                                 :mask => 'ffff:ffff:ffff:ffff::',
+                                 :ipam => IPAM::MODES[:eui64])
+      ipam = IPAM::Eui64.new(:subnet => subnet, :mac => '00:11:22:33:44:55')
+      assert_equal '2001:db8::211:22ff:fe33:4455', ipam.suggest_ip
+    end
+
+    test 'should raise an exception if prefix length is not suitable' do
+      s = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8::', :cidr => 70)
```

Comment:
`subnet` reads better than `s`, the keyboard erosion you saved on this isn't worth the lack of readability :tongue: